### PR TITLE
test: refactoring state tests

### PR DESCRIPTION
**Pull request by b00f**
## Description

This PR refactors the state tests and clean them ;)

### CODE REVIEW

**Comment by b00f** [comment]
@kehiy as requested before! Please review

File: state/execution_test.go [hunk 1]
```diff
@@ -13,136 +13,132 @@ import (
...
-	}
-	b1, c1 := td.makeBlockAndCertificate(t, 0, td.valKey1, td.valKey2, td.valKey3)
-	assert.NoError(t, td.state1.CommitBlock(b1, c1))
-	assert.NoError(t, td.state2.CommitBlock(b1, c1))
-	assert.Equal(t, td.state1.LastBlockHeight(), curHeight+1)
-
-	invSubsidyTx := tx.NewSubsidyTx(1, td.valKey2.Address(),
-		td.state1.params.BlockReward, "duplicated subsidy transaction")
+	proposer := td.state.Proposer(0)
+	lockTime := td.state.LastBlockHeight()
```
Comment (amirvalhalla) [inline state/execution_test.go:17]: `locktime` not be `LastBlockHeight + 1` ?

**Review by amirvalhalla — CHANGES_REQUESTED** [review]

File: state/execution_test.go (hunk 1 above)
Comment (b00f) [reply state/execution_test.go:17]: Technically it can be any block number in range of `[curHeight - 8640, curHeight)`

**Review by amirvalhalla — APPROVED** [review]

**Review by kehiy — APPROVED** [review]
@b00f thanks, state tests are much more readable now!
applying changes also is more easy.

**Review by amirvalhalla — APPROVED** [review]

**Pull request merged**